test/jit_hooks/test_jit_hooks.cpp

```diff
@@ -0,0 +1,270 @@
+
+  auto output = module(inputs);
+  std::cout << "----- module output: " << output << std::endl;
+  std::ostringstream stream;
+  stream << output;
+  std::string output_str =  stream.str();
+  AT_ASSERT("([pre_hook_override_name, inner_mod_name], pre_hook_override_fh)" == output_str);
+}
+
+int main(int argc, const char* argv[]) {
+  if (argc != 2) {
+    std::cerr << "usage: test_jit_hooks <path-to-exported-script-module>\n";
+    return -1;
+  }
+  const std::string path_to_exported_script_module = argv[1];
+
+  // Note: Modules loaded in this file are produced in /test/jit_hooks/model.py
+
+  std::cout << "Tesing JIT Hooks in CPP" << std::endl; 
+  std::cout << "testing: test_submodule_multiple_hooks_single_input" << std::endl;
+  test_submodule_multiple_hooks_single_input(path_to_exported_script_module);
+  std::cout << "testing: test_submodule_hook_return_nothing" << std::endl;
+  test_submodule_hook_return_nothing(path_to_exported_script_module);
+  std::cout << "testing: test_submodule_same_hook_repeated" << std::endl;
+  test_submodule_same_hook_repeated(path_to_exported_script_module);
+  std::cout << "testing: test_submodule_forward_single_input" << std::endl;
+  test_submodule_forward_single_input(path_to_exported_script_module);
+  std::cout << "testing: test_submodule_multiple_hooks_multiple_inputs" << std::endl;
+  test_submodule_multiple_hooks_multiple_inputs(path_to_exported_script_module);
+  std::cout << "testing: test_submodule_forward_multiple_inputs" << std::endl;
+  test_submodule_forward_multiple_inputs(path_to_exported_script_module);
+  
+  std::cout << "testing: test_module_forward_single_input" << std::endl;
+  test_module_forward_single_input(path_to_exported_script_module);
+  std::cout << "testing: test_module_multiple_hooks_single_input" << std::endl;
+  test_module_multiple_hooks_single_input(path_to_exported_script_module);
+  std::cout << "testing: test_module_hook_return_nothing" << std::endl;
+  test_module_hook_return_nothing(path_to_exported_script_module);
+  std::cout << "testing: test_module_same_hook_repeated" << std::endl;
+  test_module_same_hook_repeated(path_to_exported_script_module);
+  std::cout << "testing: test_module_forward_multiple_inputs" << std::endl;
+  test_module_forward_multiple_inputs(path_to_exported_script_module);
+  std::cout << "testing: test_module_multiple_hooks_multiple_inputs" << std::endl;
+  test_module_multiple_hooks_multiple_inputs(path_to_exported_script_module);
+  
+  std::cout << "testing: test_module_no_forward_input" << std::endl;
+  test_module_no_forward_input(path_to_exported_script_module);
+  std::cout << "testing: test_forward_tuple_input" << std::endl;
+  test_forward_tuple_input(path_to_exported_script_module);
+
+  std::cout << "testing: test_module_forward_invocation_no_hooks_run" << std::endl;
+  test_module_forward_invocation_no_hooks_run(path_to_exported_script_module);
+
+  std::cout << "JIT CPP Hooks okay!" << std::endl;
```

Comment:
Nit: It's probably automatically inserted but we are using the return value in a shell script sooo
```suggestion
  std::cout << "JIT CPP Hooks okay!" << std::endl;
  return 0;
```